### PR TITLE
Falco version and driver version are not coupled anymore

### LINKED ISSUE #1107: Falco 0.21.0 failed to start on GKE
**Describe the bug**

Falco 0.21.0 failed to start when eBPF is enabled trying to locate the default probe

**How to reproduce it**

Simply deploy version 0.21.0 on GKE version (1.14.10-gke.17) 

Set BPF probe variable to an empty string 

```
- name: FALCO_BPF_PROBE
  value: ""
```

**Expected behaviour**

Falco should start successfully and should locate the default probe.

**Screenshots**
```
DEBUG: kubectl logs -f falco-6pflw falco
* Setting up /usr/src links from host
* Mounting debugfs
Found kernel config at /proc/config.gz
* COS detected (build 11647.293.0), using cos kernel headers...
* Downloading https://storage.googleapis.com/cos-tools/11647.293.0/kernel-headers.tgz
* Extracting kernel sources
* Configuring kernel
* Trying to compile BPF probe falco-probe-bpf (falco-probe-bpf-latest-x86_64-4.14.138+-3ed0774cb82bc5c7f3b6f5190c3b82ef.o)
make: *** /usr/src/falco-latest/bpf: No such file or directory.  Stop.
mv: cannot stat '/usr/src/falco-latest/bpf/probe.o': No such file or directory
* Trying to download precompiled BPF probe from https://s3.amazonaws.com/download.draios.com/stable/sysdig-probe-binaries/falco-probe-bpf-latest-x86_64-4.14.138%2B-3ed0774cb82bc5c7f3b6f5190c3b82ef.o

curl: (7) Failed to connect to s3.amazonaws.com port 443: Connection timed out
* Failure to find a BPF probe
Thu Mar 19 11:36:24 2020: Falco initialized with configuration file /etc/falco/falco.yaml
Thu Mar 19 11:36:24 2020: Loading rules from file /etc/falco/falco_rules.yaml:
Thu Mar 19 11:36:25 2020: Loading rules from file /etc/falco/falco_rules.local.yaml:
Thu Mar 19 11:36:26 2020: Loading rules from file /etc/falco/rules.d/mycustom-rule-appending.yml:
Thu Mar 19 11:36:26 2020: Unable to load the driver. Exiting.
Thu Mar 19 11:36:26 2020: Runtime error: can't open BPF probe '/root/.falco/falco-probe-bpf.o': Errno 2. Exiting.
```

**Environment**

- Falco version: 0.21.0
- System info: 
```
Thu Mar 19 11:53:16 2020: Falco initialized with configuration file /etc/falco/falco.yaml
Thu Mar 19 11:53:16 2020: Loading rules from file /etc/falco/falco_rules.yaml:
Thu Mar 19 11:53:17 2020: Loading rules from file /etc/falco/falco_rules.local.yaml:
Thu Mar 19 11:53:18 2020: Loading rules from file /etc/falco/rules.d/mycustom-rule-appending.yml:
{
  "machine": "x86_64",
  "nodename": "<GKE node name>",
  "release": "4.14.138+",
  "sysname": "Linux",
  "version": "#1 SMP Tue Sep 3 02:58:08 PDT 2019"
}
```
- Cloud provider or hardware configuration: Google Cloud 
- OS: 
```
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```
- Kernel:
```
Linux <nodepoolname>-default-pool-85a2c044-b1vb 4.14.138+ #1 SMP Tue Sep 3 02:58:08 PDT 2019 x86_64 **GNU/Linux**
```
- Installation method:
Kubernetes

**Additional context**
Installed through helm with taking into consideration editing the variable for eBPF to an empty string

### LINKED ISSUE #1110: Installing Falco Issue
Hi,

So we wanted to test out this released feature: https://github.com/falcosecurity/falco/pull/1099

We followed the install instructions and ran the following commands

```
sudo rpm --import https://falco.org/repo/falcosecurity-3672BA8F.asc
sudo curl -s -o /etc/yum.repos.d/falcosecurity.repo https://falco.org/repo/falcosecurity-rpm-dev.repo
sudo yum -y install kernel-devel-$(uname -r)
sudo yum -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
sudo yum -y install falco
```

Our issue here is that when we run `yum -y install falco` it tries to install three dependencies.  2 of them get installed successfully but the `falco-0.21.0_7+7b44aaf-1.x86_64` dependency in the path `/usr/src/` does not.

We get this error:
```
Error! Could not find module source directory.
Directory: /usr/src/falco-0.21.0-7+7b44aaf does not exist.
```

So I think I ended up figuring it out.  the directory mentioned in /usr/src/ IS there but it has a different name.  Instead of it being named falco-version (in this case `falco-0.21.0-7+7b44aaf`)  its named `falco-be1ea2d9482d0e6e2cb14a0fd7e08cbecf517f94`.  I believe this is our issue.  its trying to find `/usr/src/falco-0.21.0-7+7b44aaf` when installing but for some reason, that doesn’t exist--instead the directory is `/usr/src/falco-be1ea2d9482d0e6e2cb14a0fd7e08cbecf517f94` which is why this is failing.

So it looks like whatever script pulls down falco and puts it in /usr/src/ names it incorrectly.  Is this something you guys can look into?

### PR DESCRIPTION
**Pull request by leodido**
**What type of PR is this?**

/kind bug

/kind cleanup

**Any specific area of the project related to this PR?**

/area build

/area tests

**What this PR does / why we need it**:

This PR introduces some fixes and updates needed because the Falco version and the driver version are distinct values now.

So this PR mainly updates the `falco-probe-loader` script for this reason. Also renaming it to `falco-driver-loader`.
Futhermore, this script obtains now some values (like the `DRIVER_VERSION` variable value) at configuration time.

Finally it also ensures that the `FALCO_VERSION` env variable inside docker containers contains the same value returned by `falco --version`, not the docker image tag.

**Which issue(s) this PR fixes**:

Fixes #1107
Fixes #1110 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
update: falco-probe-loader script is falco-driver-loader now
new: falco version and driver version are distinct and not coupled anymore
fix: /usr/bin/falco-${DRIVER_VERSION} driver directory
fix: FALCO_VERSION env variable inside Falco containers contains the Falco version now (not the docker image tag)
```

### CODE REVIEW

**Comment by leodido** [comment]
cc @fntlnz

File: CMakeLists.txt [hunk 1]
```diff
@@ -73,6 +73,7 @@ include(GetFalcoVersion)
 set(PACKAGE_NAME "falco")
 set(PROBE_NAME "falco-probe")
 set(PROBE_DEVICE_NAME "falco")
+set(DRIVER_LOOKUP_URL "https://s3.amazonaws.com/download.draios.com")
```
[the remark this answers is not in the document]
Comment (leodido) [reply CMakeLists.txt]: As soon we complete the drivers build grid yes :)

Clearly, we should also adapt the script internally but we're really close to close the circle! :)

File: test/falco_test.py [hunk 2]
```diff
@@ -175,7 +175,7 @@ def setUp(self):
         self.copy_local_driver = self.params.get('copy_local_driver', '*', default=False)
 
         # Used by possibly_copy_local_driver as well as docker run
-        self.module_dir = os.path.expanduser("~/.sysdig")
+        self.module_dir = os.path.expanduser("~/.falco")
```
Comment (kris-nova) [inline test/falco_test.py]: !!!

File: userspace/falco/falco.cpp [hunk 3]
```diff
@@ -624,6 +624,7 @@ int falco_init(int argc, char **argv)
 				if(string(long_options[long_index].name) == "version")
 				{
 					printf("Falco version: %s\n", FALCO_VERSION);
+					printf("Driver version: %s\n", DRIVER_VERSION);
```
Comment (kris-nova) [inline userspace/falco/falco.cpp]: So we don't need to answer this now - but we should start thinking about what/when/where we output various pieces of information in Falco

We could standardize some of this - and I think we should give meaningful information whenever possible

File: CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (kris-nova) [reply CMakeLists.txt]: I will add this to the doc for the week - I think drivers being built and hosted somewhere is critical - falco isn't really falco until the drivers are working and documented

File: userspace/falco/falco.cpp (hunk 3 above)
Comment (leodido) [reply userspace/falco/falco.cpp]: I agree.

In this case the reasoning was: Falco version is a different thing than the driver version now. Thus, we need to communicate it to the users. :)

File: test/falco_test.py (hunk 2 above)
Comment (leodido) [reply test/falco_test.py]: 😸

**Pull request merged**